sys/kern/src/arch/arm_m.rs

```diff
@@ -278,25 +279,21 @@ pub fn reinitialize(task: &mut task::Task) {
     let frame_size = core::mem::size_of::<ExtendedExceptionFrame>();
     // The subtract below can overflow if the task table is corrupt -- let's
     // make that failure a little easier to read:
-    uassert!(initial_stack as usize >= frame_size);
+    uassert!(initial_stack >= frame_size);
     // Ok. Generate a uslice for the task's starting stack frame.
     let mut frame_uslice: USlice<ExtendedExceptionFrame> =
-        USlice::from_raw(initial_stack as usize - frame_size, 1).unwrap_lite();
+        USlice::from_raw(initial_stack - frame_size, 1).unwrap_lite();
     // Before we set our frame, find the region that contains our initial stack
     // pointer, and zap the region from the base to the stack pointer with a
     // distinct (and storied) pattern.
-    for region in task.region_table().iter() {
-        if initial_stack < region.base {
-            continue;
-        }
-
-        if initial_stack > region.base + region.size {
-            continue;
-        }
-
+    if let Some(region) = task
+        .region_table()
+        .iter()
+        .find(|region| region.contains(initial_stack))
+    {
```

Comment:
There's the possibility of a behavior change here: the previous code would zap **every** region that contains the stack pointer, while the new code stops after finding one region.  It seems unlikely to matter – are there any cases where this is significant?

Reply:
No, multiple regions containing the stack pointer would have been rejected by the region overlap check in the build system. I changed this code for sort of a peripheral reason: the loop held a borrow on the region table, but my switch to using `.region_table()` caused that to be a borrow on the entire task. Inserting a `break;` into the loop would also have fixed the borrowing issue, but at that point this seemed like a clearer statement of intent.